libretro-rs/src/core_macro.rs

```diff
@@ -46,7 +45,7 @@ macro_rules! libretro_core {
       }
 
       #[no_mangle]
-      extern "C" fn retro_set_environment(cb: retro_environment_t) {
+      extern "C" fn retro_set_environment(cb: EnvironmentCallback) {
```

Comment:
Could you provide the motivation behind this change?

Reply:
> I could leave it as `retro_environment_t` and unwrap it immediately if you feel strongly about it, but I don't see the point in trying to guard against a bad frontend. There's just too many ways the frontend could pass bad data and cause undefined behavior if it really wanted to. We're already playing with fire every time we use a pointer or the environment callback. I think we may as well assume the frontend behaves correctly, just like we do with the Rust standard library.

I actually think the exact opposite here. The library should handle the potential (admittedly, edge case) ugly bits about interoping with libretro front ends. If a front end or other malicious actor wanted to send `nullptr` here, we should handle that gracefully and not bother user code with it. I think the best we can do in this case is `.expect("the argument for 'cb' cannot be a null pointer.")`, I'm also not sure how the FFI would handle a method that we've declared is never null being passed a null value. I believe it's always a good practice to validate the inputs you're being given before using them, my $0.02